Sources/SwiftASN1/Basic ASN1 Types/ObjectIdentifier.swift

```diff
@@ -143,25 +144,65 @@ extension ASN1ObjectIdentifier: Hashable {}
 
 extension ASN1ObjectIdentifier: Sendable {}
 
-extension ASN1ObjectIdentifier: ExpressibleByArrayLiteral {
+extension ASN1ObjectIdentifier {
+    /// Initializes ``ASN1ObjectIdentifier`` from its OID components
+    /// - Parameter elements: The OID components
     @inlinable
-    public init(arrayLiteral elements: UInt...) {
+    public init(elements: some Collection<UInt>) throws {
         var bytes = [UInt8]()
-        var components = elements[...]
-        guard let firstComponent = components.popFirst(), let secondComponent = components.popFirst() else {
-            preconditionFailure("Invalid number of OID components: must be at least two!")
+
+        guard elements.count >= 2 else {
```

Comment:
It might be easier to wrap this check up with getting the first two elements. That avoids a force-unwrap altogether (as it's just a guard-let), and performs better for cases where we don't have a cheap implementation of `.count` (as on `Collection` it may be `O(n)`).